content/src/main/content/jcr_root/apps/acs-commons/components/authoring/component-cloner/clientlibs/classic/js/listeners.js

```diff
@@ -0,0 +1,72 @@
+/*
+ * #%L
+ * ACS AEM Commons Bundle
+ * %%
+ * Copyright (C) 2018 Adobe
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+/**
+ * Classic UI Listener.
+ * Function to build URI and send a GET request to the servlet to clone the specified component by its path.
+ * See https://helpx.adobe.com/experience-manager/6-3/sites/developing/using/reference-materials/widgets-api/index.html?class=CQ.Dialog
+ *     - Public Events
+ *          - beforesubmit
+ * @param dialog CQ.Dialog
+ * @returns {boolean}
+ */
+var clone = function (dialog) {
```

Comment:
These functions `clone` and `callback` are not scoped at all, and have super generic names that could potentially conflict with other global vars/functions.  Should we wrap all of this code inside an anonymous function?

```
(function() {
   // code goes here
})();
```

Reply:
ah nvm I see that `clone()` is called from the dialog - that being the case, we might want to scope these fucntions inside a global "componentCloner" object or something.

```
var componentCloner = {};
componentCloner.clone = function(dialog) {
...
};
componentCloner.callback = function {
...
};
```